### ISSUE TITLE
[MU4 Issue] Time signatures do not edit when selected all

### ISSUE THREAD

**Issue by navickasm** [musescore/MuseScore#16406]
**Describe the bug**
When selecting all time signatures in a conductor's stave, when one goes to edit their properties, the edits only apply to the individual time signature that was most recently clicked.

**To Reproduce**
1. In a new project with multiple instruments, click a time signature so that all of them turn blue (i.e. selected)
2. Select "time signature properties"
3. Change the appearance to cut time

**Expected behavior**
The settings changed should apply to all time signatures, as they are all selected.

**Platform information**
 - OS: Windows

**Comment by svetter** [comment]
I've recently run into this issue as well and would like to try fixing it (first time contributor). Before I do that, could someone from @musescore give some basic direction to what the UX should look like?

I would argue that editing the signature of all instruments at once should be the default, not least since you can't even select the time signature for a single instrument. So my suggestion would be to change the existing action to editing all at once, and adding a second action to the context menu which reads "Edit [time signature] properties for this instrument" (or similar) and have it behave accordingly.

**Comment by cbjeukendrup** [comment]
> not least since you can't even select the time signature for a single instrument

You can, by holding Ctrl/Cmd while clicking on it. 

But I think that's even more reason to edit the time signature for all instruments at once in the time signature properties dialog. If the user wants to edit it only for specific instruments, they just select those individual time signatures using Ctrl/Cmd, and then the time signature properties dialog should edit only those selected ones.

The only difficulty about this arises when not all selected time signatures have the same values. This is for example the case when _local_ time signatures are present, i.e. some instruments have a different time signature than others. (These can be created by holding Ctrl/Cmd while dragging them to the score from the Palettes.) 
So for that situation, a solution should be found.

I know that @avvvvve is working on a big research/redesign project about the whole UX around time signatures, but I believe that is not finished yet (correct me if I'm wrong!). @avvvvve do you think it's still worthwhile to work on this specific fix in the meantime?

**Comment by svetter** [comment]
Thanks for correcting me, I didn't know about the Ctrl/Cmd+Click - and also for providing the context of current developments.

As for the problem with editing differing time signatures together, would it make sense to populate the dialog with the current state of the clicked instrument's time signature (or the one clicked most recently) and then just applying the new signature properties to all selected instruments when the user hits OK? Seems to me like the most intuitive solution.

**Comment by avvvvve** [comment]
@cbjeukendrup Yeah, that redesign is still in progress, but fixing the current selection/editing behavior will surely need to be done. If it's simple enough to implement, I don't see an issue with you working on this @svetter!

Here's how I'd expect it to work:

- Maintain current selection behavior for non-local time signatures (i.e. clicking selects time signatures across all instruments, Ctrl/Cmd + Click selects/deselects individual time signatures)
- Changes made in the time signature properties dialog should affect all time signatures that are visually selected in the score.
- If you've edited the properties of a non-local time signature on one instrument but not others, it should still get selected as part of the time signature "group".
- NEW: If a local time signature is added to an instrument on the same measure as a non-local time signature, clicking the local time signature should _not_ select any of the other time signatures in other instruments at the same time. Conversely, clicking the non-local time signature should select every time signature except for the local time signature. That way, you can easily edit the properties of the non-local time signature on the whole without affecting the local one. The local time signature can be added to the selection using Ctrl/Cmd + Click. 
- NEW: If there are multiple local time signatures on different instruments at the same measure, selecting one does not select the others, even if they have the same value/properties.

Sound good?

**Comment by svetter** [comment]
Yup, thanks, that should give me all I need to take a stab at it. No promises of course, I can imagine running into some complexities beyond my first-time-contributor pay grade.

**Comment by cbjeukendrup** [comment]
It is by design that editing the properties of selected time signatures in a score only affects those time signature in that score. 
However, you can Ctrl+Shift+drag a modified time signature to the palettes, and that should preserve its properties when adding that time signature to a new score.

**Comment by sammik** [comment]
@cbjeukendrup Are You sure, it is by design, that changes made in score are not applied into parts?
If so, I would expect "link/unlink" option.

Now it is, at least, inconsistent, some properties are linked (signature text), but others are not (beam gruping).


https://github.com/user-attachments/assets/fa0e378b-fd24-4468-bd77-baf232e3cf9b

**Comment by oktophonie** [comment]
For most properties, by default, changes made in the score will affect that item in the parts; changes made in the parts will affect only that part, desychronising either the position or the style/appearance of the item from the score. Some properties are not desychronisable so changes made in any place will affect all instances.

Until https://github.com/musescore/MuseScore/pull/24014, no changes made to time signature properties in the score weren't propagating to parts at all. Changes to text now do, but changes to beam groupings still don't seem to be, which is a bug for sure. Whether @svetter wants to look at that is up to him!

But the other question here is whether time signature properties _should_ be desynchronisable between score and parts in the first place. I can maybe imagine some rather exotic niche use cases for doing it with the text, but with the beam groupings..?

**Comment by SteveBlower** [comment]
> I can maybe imagine some rather exotic niche use cases for doing it with the text, but with the beam groupings..?

Here is an imagined case for desynchronised beaming:

A score might have the vocal parts beamed to match instrumental parts with the same rhythm and so make that similarity more obvious, but the individual vocal parts would be un-beamed for singers that are more familiar with that convention?

I can't say I have ever seen that but I can see a potential advantage.

**Comment by svetter** [comment]
It's difficult because you'd probably want to keep the option open but not make it confusing and unintuitive for everyone else. That would mean keeping time signatures completely in sync across scores by default but having some method of breaking sync. How would that look like though? A checkbox in the dialog?

**Comment by wizofaus** [comment]
> 
> A score might have the vocal parts beamed to match instrumental parts with the same rhythm and so make that similarity more obvious, but the individual vocal parts would be un-beamed for singers that are more familiar with that convention?

I'd say the only times this happens is when a conductor is using a more modern edition of the full score that uses metrical beaming, whereas vocalists are stuck with older editions of their vocal scores that use syllabic beaming.  It's never a desirable outcome.

**Comment by svetter** [comment]
Can we get an official adjudication from @musescore on the question of synchronisation between main score and parts?

**Comment by oktophonie** [comment]
@svetter Beam groupings should synchronise between score and parts.
Appearance can be allowed to desynchronise (as it does currently), even though I can only imagine this being useful in very specialised circumstances.

**Comment by MRoth1910** [comment]
This has been sitting for a while and I believe it's essentially my issue with vocal scores which naturally have multiple parts (so I did not open a new issue). When creating Renaissance polyphonic scores, you might decide to create the score with a given meter but need to change the appearance for practical reasons, to be more historically accurate, or some other reason.

1. Right-click a time signature, then right-click -> Select -> Same subtype.
2. Properties -> Time Signature Properties, then change the appearence as desired.
3. Only the first time signature changes, instead of all 4.

This is tedious to change, particularly in works that change meters (usually from some form of double to triple meter and back); whatever change I make in one place, if I select Same subtype, then I really would prefer to change everythng all at once!

<img width="990" height="322" alt="Image" src="https://github.com/user-attachments/assets/0ef08015-e8d2-4140-adec-8ffec8f6239c" />

<img width="990" height="322" alt="Image" src="https://github.com/user-attachments/assets/3c726d87-3375-4787-a93a-f3883bb59b64" />

**Comment by MarcSabatella** [comment]
Just a note, you can Ctrl+Shfit+drag a customized time signature back to the palette for easy reuse - no need to keep changing properties manually each and every time.

**Comment by MRoth1910** [comment]
That isn't the problem.

**Comment by MarcSabatella** [comment]
Hmm, seems you might be misunderstanding my helpful suggestion.  It's an enormous time saver, one that your comment seems to indicate you are not aware of.  And your comment did require clarification, as it might mislead someone into thinking that the laborious workaround you allude is actually necessary when in fact it is not, because one can use the custom palette item to eliminate all the extra properties settings - you literally need to do it only once per time signature (not once per staff, or once per change - literally once per time signature).

**Comment by MRoth1910** [comment]
If **_multiple objects_** are selected in blue, they should change. **_Every other thing changes when multiple items are selected at once_**

Plus, there's no way to tell the difference in the palette between the 4/2 that has a Renaissance tempus imperfectus, prolatio minor sign and the 2/2 that does (or indeed any other meter that I might choose to represent with the same mensuration sign).

anyway, it is not at all obvious (as apparently multiple people don't understand how customizing appearence works, based on what I read here). and because I'm on a Mac, where the multiple-instances problem is not yet solved, the use of the palette is not synced between projects, so it's sort of fundamentally useless.
